tests/run-make/output-with-hyphens/rmake.rs

```diff
@@ -0,0 +1,17 @@
+// Rust files with hyphens in their filename should
+// not result in compiled libraries keeping that hyphen -
+// it should become an underscore. Only bin executables
+// should keep the hyphen. This test ensures that this rule
+// remains enforced.
+// See https://github.com/rust-lang/rust/pull/23786
+
+//@ ignore-cross-compile
+
+use run_make_support::{bin_name, path, rustc};
+
+fn main() {
+    rustc().input("foo-bar.rs").crate_type("bin").run();
+    assert!(path(bin_name("foo-bar")).exists());
+    rustc().input("foo-bar.rs").crate_type("lib").run();
+    assert!(path(bin_name("libfoo_bar.rlib")).exists());
```

Comment:
This seems wrong. This should be `rust_lib_name("foo_bar")`, right?

Reply:
It is technically the same thing since `rust_lib_name` only does `format!("lib{name}.rlib")`, but for the sake of consistency I will change it.

Reply:
I mean, it returns the same contents in this specific case, but it's semantically wrong, and would probably fail on Windows anyway, since it would add `.exe` at the end :)

Reply:
(You didn't just hardcode the name, but also put it inside `bin_name`, not sure if you have noticed that before. Anyway, it's fixed now :) )

Reply:
Oh wow, I did *not* notice that. Thank you.